### PR TITLE
simplified threshold creation and message reconstruction

### PR DESCRIPTION
**Pull request by liorrutenberg**
Uses BLS lib to create shares and reconstruct the message

### CODE REVIEW

File: cli/threshold.go [hunk 1]
```diff
@@ -30,8 +30,8 @@ var createThresholdCmd = &cobra.Command{
 		if err := baseKey.SetHexString(privKey); err != nil {
 			Logger.Fatal("failed to set hex private key", zap.Error(err))
 		}
-
-		privKeys, err := threshold.Create(baseKey.Serialize(), keysCount)
+		// https://github.com/ethereum/eth2-ssv/issues/22
+		privKeys, err := threshold.Create(baseKey.Serialize(), keysCount-1, keysCount)
```
Comment (alonmuroch) [inline cli/threshold.go:37]: @lior-blox maybe we should add a comment on why it's -1

File: utils/threshold/threshold_test.go [hunk 2]
```diff
@@ -1,31 +1,116 @@
...
 	}
 
 	// reconstruct
-	sig, _ := ReconstructSignatures(m)
-	originalSig := bls.Sign{}
+	sig, _ := ReconstructSignatures(sigVec)
+	require.True(t, originalSig.IsEqual(sig))
 	require.NoError(t, originalSig.Deserialize(sig.Serialize()))
-	require.True(t, originalSig.VerifyByte(sk.GetPublicKey(), []byte("hello")))
+	require.True(t, originalSig.VerifyByte(sk.GetPublicKey(), message))
```
Comment (alonmuroch) [inline utils/threshold/threshold_test.go]: @lior-blox we should add a test for invalid partial sigs to make sure it's actually working.

**Review by alonmuroch — CHANGES_REQUESTED** [review]

File: cli/threshold.go (hunk 1 above)
Comment (liorrutenberg) [reply cli/threshold.go:37]: @alonmuroch added

File: utils/threshold/threshold_test.go (hunk 2 above)
Comment (liorrutenberg) [reply utils/threshold/threshold_test.go]: @alonmuroch done

**Pull request merged**